shell/config/router/navigation-guards/attempt-first-login.js

```diff
@@ -0,0 +1,71 @@
+import { SETUP } from '@shell/config/query-params';
+import { SETTING } from '@shell/config/settings';
+import { MANAGEMENT, NORMAN } from '@shell/config/types';
+import { tryInitialSetup } from '@shell/utils/auth';
+import { routeNameMatched } from '@shell/utils/router';
+
+export function install(router, context) {
+  router.beforeEach((from, to, next) => attemptFirstLogin(from, to, next, context));
+}
+
+export async function attemptFirstLogin(from, to, next, { store }) {
+  if (routeNameMatched(to, 'unauthenticated')) {
+    return next();
+  }
+
+  // Initial ?setup=admin-password can technically be on any route
+  let initialPass = to.query[SETUP];
+  let firstLogin = null;
+
+  try {
+    const res = store.getters['management/byId'](MANAGEMENT.SETTING, SETTING.FIRST_LOGIN);
+    const plSetting = store.getters['management/byId'](MANAGEMENT.SETTING, SETTING.PL);
+
+    firstLogin = res?.value === 'true';
+
+    if (!initialPass && plSetting?.value === 'Harvester') {
+      initialPass = 'admin';
+    }
+  } catch (e) {
```

Comment:
This is a part of the original implementation. I think we'll want to fix this if we get around to breaking this apart further.